### PR TITLE
Added `use_activation_hooks: bool` to swap

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #214

This PR exposes `use_activation_hooks` to `swap_linear_with_float8_linear`. I am not sure if we want this long-term. Are we going to consolidate to _always_ using activation hooks in the future (because we need it for DTensor (?))?

**Test Plan**
```
./test/test_everything.sh
```

Differential Revision: [D53780333](https://our.internmc.facebook.com/intern/diff/D53780333)

### CODE REVIEW

File: float8_experimental/float8_linear_utils.py [hunk 1]
```diff
@@ -70,8 +70,10 @@ def _update_history_with_new_amax(new_amax, amax_history):
 def swap_linear_with_float8_linear(
     module: nn.Module,
     module_cls: Type[nn.Module],
-    emulate: bool = False,
+    *,
```
Comment (awgu) [inline float8_experimental/float8_linear_utils.py:73]: I was thinking that forcing these to be kwarg only would be less error prone.

**Comment by awgu** [comment]
@awgu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53780333).

File: float8_experimental/float8_linear_utils.py (hunk 1 above)
Comment (drisspg) [reply float8_experimental/float8_linear_utils.py:92]: yeah I like it!

**Review by drisspg — APPROVED** [review]

**Comment by awgu** [comment]
@drisspg I am seeing:
```
[rank1]:   File "/data/users/andgu/float8_experimental/float8_experimental/float8_tensor.py", line 284, in __torch_dispatch__
[rank1]:     raise NotImplementedError(f"attempting to run {func}, this is not supported")
```
from `test_dtensor.py` with today's `pytorch` `viable/strict` (2ad66e6bf087a7fde1dd4fcfa14e99f9b7bc1440). It should be unrelated to this PR though.

**Comment by awgu** [comment]
@awgu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53780333).

**Pull request closed without merging**